client/me/purchases/credit-cards/credit-card-delete.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+
+import React, { FunctionComponent, useState, useCallback } from 'react';
+import { connect } from 'react-redux';
+import { useTranslate, TranslateResult } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { deleteStoredCard } from 'state/stored-cards/actions';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { isDeletingStoredCard } from 'state/stored-cards/selectors';
+import { Button } from '@automattic/components';
+import { isPaymentAgreement } from 'lib/checkout/payment-methods';
+import StoredCard from 'components/credit-card/stored-card';
+import PaymentMethodDeleteDialog from './payment-method-delete-dialog';
+
+/**
+ * Style dependencies
+ */
+import './credit-card-delete.scss';
+
+interface Card {
+	card?: string;
+	email: string;
+	card_type?: string;
+	payment_partner: string;
+	name: string;
+	expiry: string;
+	stored_details_id: string;
+}
+
+interface Props {
+	card: Card;
+	errorNotice: ( msg: TranslateResult ) => void;
+	successNotice: ( msg: TranslateResult ) => void;
+	deleteStoredCard: ( card: Card ) => Promise< void >;
+	isDeleting: boolean;
+}
+
+const CreditCardDelete: FunctionComponent< Props > = props => {
+	const translate = useTranslate();
+	const [ isDialogVisible, setIsDialogVisible ] = useState( false );
+
+	const closeDialog = useCallback( () => setIsDialogVisible( false ), [] );
+
+	const handleDelete = useCallback( () => {
+		closeDialog();
+		props
+			.deleteStoredCard( props.card )
+			.then( () => {
+				if ( isPaymentAgreement( props.card ) ) {
+					props.successNotice( translate( 'Payment method deleted successfully' ) );
+				} else {
+					props.successNotice( translate( 'Card deleted successfully' ) );
+				}
+			} )
+			.catch( ( error: Error ) => {
+				errorNotice( error.message );
+			} );
+	}, [ closeDialog, props, translate ] );
```

Comment:
Having `props` as a dependency is fine here because I think `connect` makes this a pure component, but if (as I suggest below) you remove `connect`, you'd need to make sure to use `React.memo()` on this component before exporting. Could you instead just have the specific props you need listed as dependencies? It would have the added benefit of making this component less likely to do something unexpected in the future if the props do change.